gammapy/utils/coordinates/fov.py

```diff
@@ -1,7 +1,91 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
-from astropy.coordinates import SkyCoord, SkyOffsetFrame
+import astropy.units as u
+import numpy as np
+from astropy.coordinates import (
+    AltAz,
+    Angle,
+    BaseCoordinateFrame,
+    CoordinateAttribute,
+    DynamicMatrixTransform,
+    EarthLocationAttribute,
+    FunctionTransform,
+    RepresentationMapping,
+    SkyCoord,
+    SkyOffsetFrame,
+    TimeAttribute,
+    UnitSphericalRepresentation,
+    frame_transform_graph,
+)
+from astropy.coordinates.matrix_utilities import matrix_transpose, rotation_matrix
 
-__all__ = ["fov_to_sky", "sky_to_fov"]
+__all__ = ["FoVFrame", "fov_to_sky", "sky_to_fov"]
+
+reflect_lon_matrix = np.array([[1, 0, 0], [0, -1, 0], [0, 0, 1]])
+
+
+class FoVFrame(BaseCoordinateFrame):
+    """
+    FoV coordinate frame. Centered on `origin` and aligned on AltAz frame at `location` and `obstime`.
+
+    Longitudes are reversed.
+
+    Attributes
+    ----------
+    origin: astropy.coordinates.SkyCoord[AltAz]
+        Origin of this frame as a HorizonCoordinate
+    obstime: astropy.time.Time
+        Observation time
+    location: astropy.coordinates.EarthLocation
+        Location of the telescope
+    """
+
+    frame_specific_representation_info = {
+        UnitSphericalRepresentation: [
+            RepresentationMapping("lon", "fov_lon"),
+            RepresentationMapping("lat", "fov_lat"),
+        ]
+    }
+    default_representation = UnitSphericalRepresentation
+
+    origin = CoordinateAttribute(default=None, frame=AltAz)
+
+    obstime = TimeAttribute(default=None)
+    location = EarthLocationAttribute(default=None)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # make sure telescope coordinate is in range [-180°, 180°]
+        if isinstance(self._data, UnitSphericalRepresentation):
+            self._data.lon.wrap_angle = Angle(180, unit=u.deg)
+
+
+@frame_transform_graph.transform(FunctionTransform, FoVFrame, FoVFrame)
+def fov_to_fov(from_fov_coord, to_fov_frame):
+    """Transform between two skyoffset frames."""
+    intermediate_from = from_fov_coord.transform_to(from_fov_coord.origin)
+    intermediate_to = intermediate_from.transform_to(to_fov_frame.origin)
+    return intermediate_to.transform_to(to_fov_frame)
+
+
+@frame_transform_graph.transform(DynamicMatrixTransform, AltAz, FoVFrame)
+def altaz_to_fov(altaz_coord, fov_frame):
```

Comment:
I do not understand this function... Beyond the fact that there is no docstring and I do not know what it returns, the argument `altaz_coord` is never used!

Reply:
See https://docs.astropy.org/en/stable/coordinates/frames.html#defining-a-new-frame

Reply:
Please read the above documentation. All functions in the transformation graph have the signature:

```
def transform(from_coord, to_frame):
   ...
```

Reply:
This function will be called internally when, in this case:

```
altaz_coord = SkyCoord(..., frame=AltAz(...))
fov_coord = altaz_coord.transform_to(FoVFrame(...))
```

and it will always pass the two arguments.